cmw.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmw
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+type Serialization uint
+
+const (
+	JSONArray = Serialization(iota)
+	CBORArray
+	CBORTag
+	Unknown
+)
+
+// a CMW object holds the internal representation of a RATS conceptual message
+// wrapper
+type CMW struct {
+	typ Type
+	val Value
+	ind Indicator
+}
+
+func (o *CMW) SetMediaType(v string)     { _ = o.typ.Set(v) }
+func (o *CMW) SetContentFormat(v uint16) { _ = o.typ.Set(v) }
+func (o *CMW) SetTagNumber(v uint64)     { _ = o.typ.Set(v) }
+func (o *CMW) SetValue(v []byte)         { _ = o.val.Set(v) }
+func (o *CMW) SetIndicators(indicators ...Indicator) {
+	var v Indicator
+
+	for _, ind := range indicators {
+		v.Set(ind)
+	}
+
+	o.ind = v
+}
+
+func (o CMW) GetValue() []byte        { return o.val }
+func (o CMW) GetType() string         { return o.typ.String() }
+func (o CMW) GetIndicator() Indicator { return o.ind }
+
+// Deserialize a CMW
+func (o *CMW) Deserialize(b []byte) error {
+	switch sniff(b) {
```

Comment:
ok, understood from here, but we should have directly used the Spec numbers to map to JSONArray and CBOR Array, instead of creating another level of indirection in the code ?

Reply:
not sure I understand what "level of indirection" you are referring to?